contracts/DomainRegistry.sol

```diff
@@ -11,55 +16,81 @@ contract DomainRegistry {
 
     mapping(string => Domain) public domains;
 
-    event DomainCreated(string indexed topLevelDomain, uint256 deposit, address indexed owner);
-    event DomainReleased(string indexed topLevelDomain);
-
+    event DomainCreated(string indexed domainName, uint256 deposit, address indexed owner);
+    event DomainReleased(string indexed domainName);
 
     modifier hasRequiredDeposit() {
         require(msg.value == REQUIRED_DEPOSIT, "Wrong eth amount");
         _;
     }
 
-    modifier domainDoesNotExist(string memory _topLevelDomain) {
-        require(!domains[_topLevelDomain].isRegistered, "Domain exists");
+    modifier domainExistence(string memory domainName, bool shouldExist) {
+        bool exists = domains[domainName.stripProtocol()].isRegistered;
```

Comment:
`isRegistered` is not mandatory to save. Address is not zero means domain is registered.

Reply:
check owner